GCRCatalogs/register.py

```diff
@@ -558,12 +590,22 @@ def get_root_dir():
     return _config_register.root_dir
 
 
-def set_root_dir(path):
+def set_root_dir(path, write_to_config=False):
     """
     Sets runtime root_dir to *path*.
+
+    Returns
+    -------
+    If write_to_config, return old persisted value (which may be None);
+    else None.
```

Comment:
See my earlier comment on `persist_root_dir` -- I think a setter function should return None or the new value. Returning the old value would be a rather unexpected behavior.